src/main/java/net/jonathangiles/tools/teenyhttpd/TeenyApplication.java

```diff
@@ -0,0 +1,187 @@
+package net.jonathangiles.tools.teenyhttpd;
+
+import net.jonathangiles.tools.teenyhttpd.annotations.*;
+import net.jonathangiles.tools.teenyhttpd.model.MessageConverter;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static net.jonathangiles.tools.teenyhttpd.model.Method.*;
+
+public class TeenyApplication {
+    private static TeenyApplication instance;
+
+    public static TeenyApplication start() {
+        if (instance == null) {
+            instance = new TeenyApplication();
+        }
+
+        instance._start();
+
+        return instance;
+    }
+
+    public static void stop() {
+        if (instance != null) {
+            instance.server.stop();
+            instance.started.set(false);
+        }
+    }
+
+    private final TeenyHttpd server;
+    private final AtomicBoolean started = new AtomicBoolean(false);
+    private final Map<String, MessageConverter> messageConverterMap;
+
+    private TeenyApplication() {
+        server = new TeenyHttpd(Integer.parseInt(System.getProperty("server.port", "8080")));
+        this.messageConverterMap = new HashMap<>();
+    }
+
+    public TeenyApplication registerMessageConverter(MessageConverter messageConverter) {
+        messageConverterMap.put(messageConverter.getContentType(), messageConverter);
+        return this;
+    }
```

Comment:
It would be pretty cool to have annotations that can be put onto message converters, e.g. `@MessageConverter` that TeenyApplication could just find at runtime. Same with `@Controller` or similar for controller classes to be found at runtime by annotation scanning.